sdk/python/kfp/dsl/artifact.py

```diff
@@ -98,129 +46,105 @@ class Artifact(object):
      their components.
   2. At runtime, Artifact objects provide helper function/utilities to access
      the underlying RuntimeArtifact pb message, and provide additional layers
-     of validation to ensure type compatibility.
+     of validation to ensure type compatibility for fields specified in the 
+     instance schema.
   """
 
-  # Name of the Artifact type.
-  TYPE_NAME = 'Artifact'
-  # Property schema.
-  # Example usage:
-  #
-  # PROPERTIES = {
-  #   'span': Property(type=PropertyType.INT),
-  #   # Comma separated of splits for an artifact. Empty string means artifact
-  #   # has no split.
-  #   'split_names': Property(type=PropertyType.STRING),
-  # }
-  PROPERTIES = None
+  TYPE_NAME="Artifact"
+
+  # Dict holding the instance's schema fields used for type checking
+  _metadata_fields = Dict[str, MetadataFieldType]
 
   # Initialization flag to support setattr / getattr behavior.
   _initialized = False
 
+  def _parse_instance_schema(self):
+    """ Parses the instance schema of the Artifact and sets up self._metadata_fields map for type-checking.
+    Expects _instance_schema to be set.
+    """
+
+    if not self._instance_schema:
+      raise RuntimeError("instance_schema not set for the Artifact")
+
+    if 'title' not in self._instance_schema:
+      raise ValueError('Invalid instance_schema, title must be present. Got: {}'.format(self._instance_schema))
+
+    schema = yaml.full_load(self._instance_schema)
+    self.TYPE_NAME = schema['title']
+    self._metadata_fields = {}
+    if 'properties' in schema.keys():
```

Comment:
For my education, is `properties` for back compatibility? what about `custom_properties`?

Reply:
This properties is the open API Schema concept to define the schema of the object. This is not related to the Artifact `properties`.

Reply:
Got it, thanks!